integration-tests/src/test/SystemApi.spec.ts

```diff
@@ -0,0 +1,43 @@
+// [object Object]
+// SPDX-License-Identifier: Apache-2.0
+
+import { ApiPromise, Keyring, WsProvider } from '@polkadot/api';
+
+describe('System API', (): void => {
+  let api: ApiPromise;
+  let alice: KeyringPair;
+
+  beforeEach(async () => {
+    process.env.NODE_ENV = 'test';
+
+    const provider = new WsProvider('ws://127.0.0.1:9944');
+
+    api = await ApiPromise.create({ provider });
+
+    const keyring = new Keyring({
+      type: 'sr25519'
+    });
+    const alice = keyring.addFromUri('//Alice', { name: 'Alice' });
+  });
+
+  afterEach(async () => {
+    await api.disconnect();
+  });
+
+  it('rpc.system.chain() works', (): void => {
+    return api.rpc.system.chain().then(result => {
+console.log(`**** DEBUG: chain=${result}===`);
+      expect(result).toEqual("Development");
```

Comment:
Disregarding the warning messages about duplicate package versions this is what I get here:

Start with `yarn test src/test/SystemApi.spec.ts`

```
  console.log
    **** DEBUG: chain=Development===

      at src/test/SystemApi.spec.ts:29:9

  console.warn
    2022-03-21 17:51:17        API/INIT: RPC methods not decorated: creditcoin_eventsSubscribe, creditcoin_eventsUnsubscribe, creditcoin_getEvents

      at apply (node_modules/@polkadot/api/node_modules/@polkadot/util/logger.cjs:81:22)
      at Object.warn (node_modules/@polkadot/api/node_modules/@polkadot/util/logger.cjs:161:14)
      at ApiPromise._filterRpcMethods (node_modules/@polkadot/api/base/Decorate.cjs:311:9)
      at ApiPromise._filterRpc (node_modules/@polkadot/api/base/Decorate.cjs:274:10)
      at ApiPromise._metaFromChain (node_modules/@polkadot/api/base/Init.cjs:350:10)
      at ApiPromise._loadMeta (node_modules/@polkadot/api/base/Init.cjs:278:229)

  console.log
    **** You are connected to chain Development using Creditcoin Node v2.0.0-beta.3-8bd1c3a-x86_64-linux-gnu

      at Object.<anonymous> (src/test/SystemApi.spec.ts:41:13)

 FAIL  src/test/SystemApi.spec.ts (6.347 s)
  System API
    ✕ rpc.system.chain() works (802 ms)
    ✓ can retrieve info via RPC (524 ms)

  ● System API › rpc.system.chain() works

    expect(received).toEqual(expected) // deep equality

    Expected: "Development"
    Received: serializes to the same string

      28 |     return api.rpc.system.chain().then(result => {
      29 | console.log(`**** DEBUG: chain=${result}===`);
    > 30 |       expect(result).toEqual("Development");
         |                      ^
      31 |     });
      32 |   });
      33 |

      at src/test/SystemApi.spec.ts:30:22

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 passed, 2 total
Snapshots:   0 total
Time:        6.418 s, estimated 7 s
Ran all test suites matching /src\/test\/SystemApi.spec.ts/i.
Force exiting Jest: Have you considered using `--detectOpenHandles` to detect async operations that kept running after all tests finished?
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I don't know what "serializes to the same string" means but neither `.toBe()` nor `.toEqual()` assertion functions work. At the same time the debug print at the top of the snippet shows that the string has the value of `Development`. 

@jfrank-summit, @nathanwhit any idea what's going on here ?

Reply:
My guess is that `result` is not a string, it's a wrapper around a string due to the encoding/decoding happening in the rpc call (I'm pretty sure it's of type `Text`, which is indeed a wrapper around a string). So when you assert that it equals `"Development"`, a `string`, the type mismatch means they aren't quite equal, even though if you serialize `result` it "serializes to the same string".

Maybe you can try checking if `result.toString()` is equal to "Development"? I would expect that to work